### PR TITLE
Day9

### PR DESCRIPTION
**Pull request by JustineSara**

### CODE REVIEW

File: src/t/d9.clj [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    [clojure.string :as cljstr]
+    [clojure.core.match :refer [match]]
+    [clojure.set :as cljset]))
+
+(def sample
+"2333133121414131402")
+
+(defn parse-input
+  [input]
+  (map (comp parse-long str) input)
```
Comment (gaverhae) [inline src/t/d9.clj:13]: Nice!

File: src/t/d9.clj [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+       (map (fn [{:keys [pos-min pos-max file-id]}]
+              (reduce + 0 (map (fn [pos] (* pos file-id)) (range pos-min (inc pos-max))))))
+       (reduce + 0)))
+
+(defn d9p1
+  [input]
+  (let [x (parse-input input)
+        {:keys [free occupied]} (space x)
+        free (sort-by :pos-min free)
+        occupied (sort-by :pos-min > occupied)]
```
Comment (gaverhae) [inline src/t/d9.clj:53]: I didn't know about that. Thanks!

File: src/t/d9.clj [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            (let [new-file-max (- (:pos-max to-be-moved) (:size to-be-filled))
+                  moved-file {:file-id (:file-id to-be-moved)
+                              :pos-min (:pos-min to-be-filled)
+                              :pos-max (:pos-max to-be-filled)
+                              :file-size (:size to-be-filled)}
+                  left-over-file (-> to-be-moved
+                                     (assoc :pos-max new-file-max)
+                                     (update :file-size - (:size to-be-filled))) ]
+              (recur (rest f-space) (sort-by :pos-min > (conj (rest o-space) moved-file left-over-file))))
+            :else (prn ["case not done?!" :size-diff size-diff :to-be-moved to-be-moved :to-de-filled to-be-filled])))))))
```
Comment (gaverhae) [inline src/t/d9.clj]:
```suggestion
            :else (prn ["case not done?!" :size-diff size-diff :to-be-moved to-be-moved :to-be-filled to-be-filled])))))))
```

**Review by gaverhae — APPROVED** [review]

File: src/t/d9.clj (hunk 2 above)
Comment (JustineSara) [reply src/t/d9.clj:53]: oh! :)
you're welcome

File: src/t/d9.clj (hunk 1 above)
Comment (JustineSara) [reply src/t/d9.clj:13]: thank you :)

**Pull request merged**